### DISCUSSION TITLE
github recovery codes

### DISCUSSION THREAD

**Discussion by spk121** [community/community#70281] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

So I have a list of 10 login codes that Github sent me in 'github-recovery-codes.txt'. 

Do I have to request a new set of codes after I log in 10 times? Or can I just keep using these codes forever?

**Comment by andrej-kocijan** [comment, marked as answer]
Once a recovery code is used it can not be used again. Note that recovery codes should be a fallback mechanism not a primary login method. You can read more about managin your recovery codes and requesting new ones [here](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication-recovery-methods)

**Comment by ktauchathuranga** [comment]
> ### Select Topic Area
> Product Feedback
> 
> ### Body
> So I have a list of 10 login codes that Github sent me in 'github-recovery-codes.txt'.
> 
> Do I have to request a new set of codes after I log in 10 times? Or can I just keep using these codes forever?

you have to request new codes 🖤

**Comment by ouchSumatra** [comment]
This does not answer the question why so many recovery codes are failing. I have tried from Microsofts site, used github as a recovery option and they are having the same problem. None of the over ten codes sent were recognized when pasted. I tried filling out a support trouble ticket and its trying to tell me what to say failing miserably addressing the problem. My prime email address lr2200(OuchSumatra) Their is no clean uncensored way to file a valid complain because everyone has gotten tired of listening, how sad?
